backend/services/auth.go

```diff
@@ -0,0 +1,59 @@
+package services
+
+import (
+	"backend/config"
+	"backend/models"
+	"backend/stores"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type (
+	AuthService interface {
+		GenerateAccessToken(username string, password string, admin bool) (accessToken string, exp int64, err error)
+		GetUserById(id string) *models.User
+	}
+
+	authService struct {
+		*config.Config
+		*stores.Stores
+	}
+)
+
+func (s *authService) GenerateAccessToken(username string, password string, admin bool) (accessToken string, exp int64, err error) {
+
+	user, err := s.User.GetByUsername(username)
+	if err != nil || (bcrypt.CompareHashAndPassword([]byte(user.PasswordHash), []byte(password)) != nil) {
+		return "", 0, err
+	}
+
+	expired := time.Now().Add(time.Hour * 72)
+
+	claims := &config.JwtCustomClaims{
+		ID:    user.ID,
+		Admin: admin,
+		RegisteredClaims: jwt.RegisteredClaims{
+			ExpiresAt: jwt.NewNumericDate(expired),
+		},
+	}
+
+	exp = expired.Unix()
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+
+	accessToken, err = token.SignedString([]byte(s.Auth.AccessSecret))
+	if err != nil {
+		return "", 0, err
+	}
+
+	return accessToken, exp, err
```

Comment:
[want]
おそらく以下の方が適切です。

```suggestion
	return accessToken, exp, nil
```

Reply:
こちらも修正完了です！